jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolPathSelector.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.tools
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.components.panels.Wrapper
+import com.intellij.util.text.nullize
+import com.intellij.util.ui.components.BorderLayoutPanel
+import software.aws.toolkits.resources.message
+import java.awt.event.ActionListener
+import java.nio.file.Path
+import java.nio.file.Paths
+import javax.swing.JButton
+
+open class ToolPathSelector<T : ToolType<*>>(private val type: T, private val validationResultHolder: Wrapper) : BorderLayoutPanel() {
+    private val testButton = JButton("Check")
+    private val pathSelector = TextFieldWithBrowseButton(null as ActionListener?)
+    private var autoDetectedPath = ""
+
+    init {
+        pathSelector.addBrowseFolderListener(
+            "Select Tool",
+            null,
+            null,
+            FileChooserDescriptorFactory.createSingleFileDescriptor(),
+        )
+
+        testButton.addActionListener {
+            val pathToCheck = getConfiguredPath() ?: autoDetectedPath
+            val label = when (val validity = checkExecutable(Paths.get(pathToCheck), type)) {
+                is Validity.Valid -> {
+                    JBLabel("<html>${type.displayName} version is ${validity.version.displayValue()}")
```

Comment:
unterminated `<html>`. are we only using html for the <br>?

Reply:
Should not be html on this one, only error

Reply:
Fixed